ompi/mca/osc/ucx/osc_ucx_component.c

```diff
@@ -507,6 +516,13 @@ static int component_select(struct ompi_win_t *win, void **base, size_t size, in
          * we don't want to initialize in the component_init()
          */
 
+        if (NULL == mca_osc_ucx_component.wpool->ucp_ctx) {
+            ret = ucp_context_init(mca_osc_ucx_component.enable_mpi_threads,  ompi_proc_world_size());
```

Comment:
Don't we have the guarantee that `ucp_ctx` is set from the call in `component_init`?

Reply:
No, inside ```component_select()```, ```ucp_ctx``` would be NULL if pml ucx calls ```opal_common_get_ucx_support_level(ctx)```  (or ```opal_common_ucx_support_level(ctx)``` in the current main branch) before osc calls it. In other words, if ```opal_common_ucx_support_level``` is already set in pml, we avoid setting it again in osc.